src/overlays/actors/ovl_En_Zl2/z_en_zl2.c

```diff
@@ -27,265 +101,1664 @@ const ActorInit En_Zl2_InitVars = {
+    }
+    *unk_194 = *unk_190;
+}
+
+void func_80B4EA40(EnZl2* this) {
+    s16* unk_190 = &this->unk_190;
+    f32* unk_27C = &this->unk_27C;
+
+    if (*unk_27C < 3.0f) {
+        *unk_190 = 0;
+    } else {
+        if (*unk_27C < 6.0f) {
+            *unk_190 = 1;
+        } else if (*unk_27C < 9.0f) {
+            *unk_190 = 2;
+        } else {
+            *unk_190 = 5;
+            this->unk_194 = *unk_190;
+            return;
+        }
+    }
+    *unk_27C += 1.0f;
+    this->unk_194 = *unk_190;
+}
+
+void func_80B4EAF4(EnZl2* this) {
+    s16* unk_190 = &this->unk_190;
+    f32* unk_27C = &this->unk_27C;
+
+    if (*unk_27C < 2.0f) {
+        *unk_190 = 5;
+    } else {
+        if (*unk_27C < 4.0f) {
+            *unk_190 = 2;
+        } else if (*unk_27C < 6.0f) {
+            *unk_190 = 1;
+        } else {
+            func_80B4E9B0(this);
+            return;
+        }
+    }
+    *unk_27C += 1.0f;
+    this->unk_194 = *unk_190;
+}
+
+void func_80B4EBB8(EnZl2* this) {
+    s16* unk_190 = &this->unk_190;
+    f32* unk_27C = &this->unk_27C;
+
+    if (*unk_27C < 2.0f) {
+        *unk_190 = 0;
+    } else {
+        if (*unk_27C < 4.0f) {
+            *unk_190 = 1;
+        } else {
+            *unk_190 = 2;
+            this->unk_194 = *unk_190;
+            return;
+        }
+    }
+    *unk_27C += 1.0f;
+    this->unk_194 = *unk_190;
+}
+
+void func_80B4EC48(EnZl2* this) {
+    s16* unk_190 = &this->unk_190;
+    f32* unk_27C = &this->unk_27C;
+
+    if (*unk_27C < 2.0f) {
+        *unk_190 = 2;
+    } else {
+        if (*unk_27C < 4.0f) {
+            *unk_190 = 1;
+        } else {
+            func_80B4E9B0(this);
+            return;
+        }
+    }
+    *unk_27C += 1.0f;
+    this->unk_194 = *unk_190;
+}
+
+void func_80B4ECE8(EnZl2* this, s16 arg1) {
+    this->unk_190 = arg1;
+    this->unk_194 = this->unk_190;
+}
+
+void func_80B4ED04(EnZl2* this, s16 arg1) {
+    this->unk_194 = arg1;
+}
+
+void func_80B4ED18(EnZl2* this, s16 arg1) {
+    this->unk_198 = arg1;
+}
+
+void func_80B4ED2C(EnZl2* this, GlobalContext* globalCtx) {
+    func_8002E4B4(globalCtx, &this->actor, 75.0f, 30.0f, 30.0f, 5);
+}
+
+s32 EnZl2_FrameUpdateMatrix(EnZl2* this) {
+    return SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+}
+
+CsCmdActorAction* EnZl2_GetNpcAction(GlobalContext* globalCtx, s32 idx) {
+    if (globalCtx->csCtx.state != 0) {
+        return globalCtx->csCtx.npcActions[idx];
+    }
+    return NULL;
+}
+
+void func_80B4EDB8(EnZl2* this, GlobalContext* globalCtx, s32 arg2) {
+    CsCmdActorAction* npcAction = EnZl2_GetNpcAction(globalCtx, arg2);
+    s16 actionRotY;
+
+    if (npcAction != NULL) {
+        this->actor.posRot.pos.x = npcAction->startPos.x;
+        this->actor.posRot.pos.y = npcAction->startPos.y;
+        this->actor.posRot.pos.z = npcAction->startPos.z;
+        actionRotY = npcAction->rot.y;
+        this->actor.shape.rot.y = actionRotY;
+        this->actor.posRot.rot.y = actionRotY;
```

Comment:
Any chance this matches? If it does it can probably be applied to other places in this file.
```suggestion
        this->actor.posRot.rot.y = this->actor.shape.rot.y = npcAction->rot.y;
```